src/databricks/labs/ucx/source_code/notebook.py

```diff
@@ -226,60 +231,80 @@ def read_cell_language(self, lines: list[str]) -> CellLanguage | None:
             return None
         return None
 
-    def new_cell(self, source: str) -> Cell:
-        return self._new_cell(source)
+    def new_cell(self, source: str, original_offset: int) -> Cell:
+        return self._new_cell(source, original_offset)
 
     def extract_cells(self, source: str) -> list[Cell] | None:
         lines = source.split('\n')
         header = f"{self.comment_prefix} {NOTEBOOK_HEADER}"
         if not lines[0].startswith(header):
             raise ValueError("Not a Databricks notebook source!")
 
-        def make_cell(cell_lines: list[str]):
+        def make_cell(cell_lines: list[str], start: int):
             # trim leading blank lines
-            while len(cell_lines) > 0 and len(cell_lines[0]) == 0:
-                cell_lines.pop(0)
+            new_cell_lines = cell_lines
```

Comment:
This is bug as you seem to be using Len of cell_lines down. Whatever changes you're doing to new_cell_lines also getting reflected in cell_lines as well.
For e.g,

<img width="255" alt="Screenshot 2024-04-14 at 12 34 53 PM" src="https://github.com/databrickslabs/ucx/assets/3398206/e87e15fd-faaa-48ad-877e-3bcec843ad09">

Also, I believe you just need length(number of items) in cell lines, so get length preserved in some variable.

Reply:
This is indeed a bug introduced by refactor.

Reply:
All fixed